python/examples/core/3-managing-gradients.py

```diff
@@ -0,0 +1,289 @@
+#
+# In a previous :ref:`tutorial <core-tutorial-first-steps>`, we have seen how metatensor
+# stores data and metadata together inside :py:class:`TensorBlock`; and groups multiple
+# blocks to form a full :py:class:`TensorMap`. To refresh our memory, let's load some
+# data (the radial spectrum from the :ref:`sparsity tutorial <core-tutorial-sparsity>`).
+# It is a tensor map with two dimensions for the keys; and 5 blocks:
+
+# sphinx_gallery_thumbnail_path = '../static/images/TensorBlock-Gradients.*'
+
+radial_spectrum = metatensor.load("radial-spectrum.npz")
+print(radial_spectrum)
+
+# %%
+#
+# If we look at one of the block, we can see that is contains gradients with respect to
+# ``"positions"``:
+
+block = radial_spectrum.block(species_center=7, species_neighbor=7)
+print(block)
+
+# %%
+#
+# These gradients are also stored as a :py:class:`TensorBlock`, with its own set of
+# metadata in the samples, components and properties dimensions.
+
+gradient = block.gradient("positions")
+print(gradient)
+
+# %%
+#
+# The samples are different from the values blocks (the block to which this gradient it
+# attached to): there is a first ``"sample"`` dimension, followed by a pair of indices
+# ``"structure", "atom"``.
+#
+# The ``"sample"`` dimension is always present in gradients, and indicate which of the
+# samples in the values block we are taking the gradient of. Here, the first row of the
+# gradients will contain a gradient of the first sample in the values; with respect to
+# the position of atom 4; while the last row of the gradients contains a gradient of the
+# second row of the values with respect to the position of atom 5.
+
+print(gradient.samples)
+
+# %%
+#
+# Re-using the notation from the previous tutorial, the values contain :math:`\rho_i`,
+# for a given atomic center :math:`i`.
+
+print(block.samples)
+
+# %%
+#
+# If we look a the samples for the values, we can express the four samples in this
+# gradient block as
+#
+# - :math:`\nabla_4 \rho_4`: gradient of the representation of atom 4 with respect to
+#   the position of atom 4;
+# - :math:`\nabla_5 \rho_4`: gradient of the representation of atom 4 with respect to
+#   the position of atom 5;
+# - :math:`\nabla_4 \rho_5`: gradient of the representation of atom 5 with respect to
+#   the position of atom 4;
+# - :math:`\nabla_5 \rho_5`: gradient of the representation of atom 5 with respect to
+#   the position of atom 5.
+#
+# You'll realize that some of the combinations of atoms are missing here: there is no
+# gradient of :math:`\rho_4` with respect to the positions of atom 0, 1, 2, *etc.* This
+# is another instance of the data sparsity that metatensor enable: only the non-zero
+# gradients are actually stored in memory.
+#
+# .. figure:: /../static/images/TensorBlock-Gradients.*
+#     :width: 400px
+#     :align: center
+#
+#     Visual illustration of the gradients, and how multiple gradient row/gradient
+#     samples can correspond to the same row/sample in the values.
```

Comment:
```suggestion
#     Visual illustration of the gradients. Multiple gradient row/gradient
#     samples can correspond to the same row/sample in the values.
```

Reply:
Why split the sentence here? This makes the first part of it very short IMO

Reply:
When I read this I found the first part as an introduction to the second. But we can keep them connected.